### ISSUE TITLE
Program unresponsive after multiple clicks while batch OCRing

### ISSUE THREAD

**Issue by orbittwz** [SubtitleEdit/subtitleedit#3398]
not related to my other issue, i got this before that...
after a while, if i minimize the batch convert window while OCRing, and comeback to it, it won't open and i just see the taskbar and the preview window

![image](https://user-images.githubusercontent.com/45605734/53271475-3b3a2880-36f7-11e9-8957-6fee1c2b79dd.png)

any clicks to open the window don't work, and i get the PING sound of Windows

**Comment by OmrSi** [comment]
Yes! I mentioned this before, it needs a cancel option, you can only close it from the Task manager!
https://github.com/SubtitleEdit/subtitleedit/issues/3051#issuecomment-415993678

**Comment by orbittwz** [comment]
hmm, it's not about the cancel button... (though it would be helpful to get one)
it's the program being unresponsive and i can't see the process, nor seeing what is happening..

**Comment by niksedk** [comment]
Yeah, it's because I took a shortcut... batch convert actually just calls the normal OCR UI and that cannot run in a background thread.
To fix this requires extracting/rewriting the OCR code - probably a good idea but not a quick fix.

**Comment by orbittwz** [comment]
Sure i understand!

**Comment by zorgoz** [comment]
Same issue after Visual sync. Subtitle window is updated, but can't save it anymore. Program reacts to WM_CLOSE message, as the confirmation dialog appears, but that too us unresponsive.

**Comment by niksedk** [comment]
@zorgoz: Not the same issue... please create a separate issue.

**Comment by shinigamichan** [comment]
@niksedk 

>batch convert actually just calls the normal OCR UI and that cannot run in a background thread.

If that's how it works, I don't see anything wrong with simply showing the normal OCR UI to indicate progress.

It would be nice to have a semi-autonomous mode where you could use "prompt for unknown words" without manually selecting files one by one. Ideally, you could even OCR multiple files in parallel and use locks to protect the user dictionary and names/noises list.
